go/vt/discovery/topology_watcher_test.go

```diff
@@ -248,94 +222,77 @@ func checkWatcher(t *testing.T, refreshKnownTablets bool) {
 	if refreshKnownTablets {
 		origTablet := tablet.CloneVT()
 		origTablet2 := tablet2.CloneVT()
-		if _, err := ts.UpdateTabletFields(context.Background(), tablet2.Alias, func(t *topodatapb.Tablet) error {
+		_, err := ts.UpdateTabletFields(context.Background(), tablet2.Alias, func(t *topodatapb.Tablet) error {
 			t.Hostname = tablet.Hostname
 			t.PortMap = tablet.PortMap
 			tablet2 = t
 			return nil
-		}); err != nil {
-			t.Fatalf("UpdateTabletFields failed: %v", err)
-		}
-		if _, err := ts.UpdateTabletFields(context.Background(), tablet.Alias, func(t *topodatapb.Tablet) error {
+		})
+		require.Nil(t, err, "UpdateTabletFields failed")
```

Comment:
Any reason to use require.Nil instead of require.NoError here and in a few other places? It's fine, just curious.

Reply:
Because the func has two return values, not just a single return value of type error. I'd have preferred the `NoError` form, but it doesn't work for this case.

Reply:
I meant that you can use this instead (in the other cases you're just using the `error` return value directly from the function):
```
require.NoError(t, err, "UpdateTabletFields failed")
```

AFAIK they are equivalent, but it's a little more explicit. Not a problem at all either way.